File: BOJ/[11057] 오르막 수/Baeksu/11057.cpp
#include <iostream>

using namespace std;
int N;
int dp[1000][10];

int ascent(int num);

int main() {
	cin >> N;
	for (int i = 0; i < 10; i++) {
		dp[0][i] = 1;
	}

	cout << ascent(N);

	return 0;
}

int ascent(int n) {
	int sum = 0;
	for (int i = 1; i < N; i++) {
		dp[i][9] = 1;
		for (int j = 8; j >= 0; j--) {
			dp[i][j] = dp[i][j + 1] + dp[i - 1][j];
			dp[i][j] %= 10007;
		}
	}

	for (int i = 0; i < 10; i++) {
		sum += dp[n - 1][i];
	}
	return sum % 10007;
}

File: BOJ/[11057] 오르막 수/DDAGUE/11057.py
def main():
    N = int(input())
    cal(N)

def cal(N):
    num = [[0 for i in range(10)] for j in range(N+1)]
    num[0] = [1 for i in range(10)]
    for i in range(1, N):
        for j in range(10):
            for k in range(j, 10):
                num[i][k] += num[i-1][j]
    print(sum(num[N-1]) % 10007)

main()